### PR TITLE
Adds high-level publish shelve API support. Refactors publish and upload_files

### LINKED ISSUE #41: High-level Shelve and publish using the new API
This will use the low-level APIs from: #22 and https://github.com/sul-dlss/purl_fetcher-client/pull/40

This should be similar to how the SDR-API does a deposit.
* It should be passed a cocina and a list of added and/or changed files.
* Call UploadFiles for each file.
* get the  deposit keys from each upload and put them in the cocina.
* Use paths for files that already exist

This MAY include:
* Making the existing shelve endpoint a no-op (shelving is part of the purl-fetcher API)
* Stop generating any files on /purl (the API will do it)

Mark this PR as `[HOLD]` until it is tested by Andrew.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #41

### CODE REVIEW

File: lib/purl_fetcher/client/publish.rb [hunk 1]
```diff
@@ -4,32 +4,36 @@ module PurlFetcher
...
-      attr_reader :cocina
+      attr_reader :cocina, :file_uploads
 
       def druid
         cocina.externalIdentifier
       end
 
       def body
-        cocina.to_json
+        {
```
Comment (justinlittman) [inline lib/purl_fetcher/client/publish.rb:33]: The signature of this API method has changed to also take a map of filenames to signed ids.

File: lib/purl_fetcher/client/upload_files.rb [hunk 2]
```diff
@@ -10,7 +10,7 @@ def self.upload(file_metadata:, filepath_map:)
         new(file_metadata: file_metadata, filepath_map: filepath_map).upload
       end
 
-      # @param [Hash<String,DirectUploadRequest>] file_metadata map of relative filepaths to file metadata
+      # @param [Array<DirectUploadRequest>] file_metadata array of DirectUploadRequests for the files to be uploaded
```
Comment (justinlittman) [inline lib/purl_fetcher/client/upload_files.rb:13]: Refactor to take an array of DirectUploadRequests; the hash isn't needed.

File: spec/purl_fetcher/client/upload_files_spec.rb [hunk 3]
```diff
@@ -64,8 +64,8 @@
     end
 
     it 'returns a list of upload responses' do
-      expect(uploader.upload.count).to eq(filepath_map.count)
-      expect(uploader.upload.first).to be_a(PurlFetcher::Client::DirectUploadResponse)
+      # expect(uploader.upload.count).to eq(filepath_map.count)
+      # expect(uploader.upload.first).to be_a(PurlFetcher::Client::DirectUploadResponse)
```
Comment (aaron-collier) [inline spec/purl_fetcher/client/upload_files_spec.rb:68]: Cruft or do we want to keep these as commented out lines?

Comment (justinlittman) [reply spec/purl_fetcher/client/upload_files_spec.rb]: Want to keep. I forgot to hit save in the editor. Fixed now @aaron-collier

**Review by aaron-collier — APPROVED** [review]

**Pull request merged**